api/src/quiz/views.py

```diff
@@ -115,6 +115,39 @@ def next_question(quiz_id: int, include_correct=False):
     return json
 
 
+# Converts a list of rows of IDs and values to a map from id to value
+def mapify(rows):
+    return {r[0]: r[1] for r in rows}
+
+
+def member_quiz_statistics(member_id: int):
+    '''Returns information about all quizzes and if the given member has completed them'''
+
+    quizzes = db_session.query(Quiz).filter(Quiz.deleted_at == None).all()
+    answered_questions_per_quiz_query = (
+        db_session.query(QuizQuestion.quiz_id, func.count(func.distinct(QuizAnswer.option_id)))
+        .join(QuizAnswer, QuizQuestion.id == QuizAnswer.question_id)
+        .filter(QuizAnswer.member_id == member_id)
+        .filter(((QuizAnswer.id == None) | QuizAnswer.correct) & (QuizAnswer.deleted_at == None) & (QuizQuestion.deleted_at == None))
+        .group_by(QuizQuestion.quiz_id)
+    )
```

Comment:
Can't really wrap my head around this query, but it seems to work 😅

Reply:
It's kinda sketchy tbh. It counts the number of correct answers the member has submitted for each quiz. However, it assumes that a correct answer is never submitted more than once (this is guaranteed by the UI, but the database can contain multiple correct responses to a question by a single member).

Reply:
But it's just used for statistics, so it's probably fine :p

Reply:
No wait! I forgot I used `count(distinct(option_id))`. It should work correctly at all times. 100%. Totally. Definitely no risk of weird bugs.